src/assets/RestApi/rest-api.md

```diff
@@ -0,0 +1,268 @@
+def get_tasks():
+    return jsonify({"tasks": tasks})
+```
+This method will return all the tasks in JSON format.
+
+To retrieve a specific task:
+```python
+@app.route("/tasks/<int:task_id>", methods=["GET"])
+def get_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    return jsonify({"task": task})
+```
+We use the id of the tasks to identify a specific task since id is unique for every tasks. This method will look for the task with provided id and return the task with the matching id. If the task with the provided id is not found, it will return the error in JSON response indicating the task was not found, with status code `404`.
+
+### POST Request
+To add a new task, implement a route that handles POST requests:
+```python
+@app.route("/tasks", methods=["POST"])
+def create_task():
+    new_task = request.get_json()
+    tasks.append(new_task)
+    return jsonify({"task": new_task}), 201
+```
+`request.get_json()` retrieves the JSON data from the request body and adds the new task to the list of tasks. After the new task has been added to the list of tasks it will return the newly created task as a JSON response with `201` status code, indicating that the task was successfully created.
+
+### PUT Request
+To update a task, implement a route that handles PUT requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["PUT"])
+def update_task(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    task.update(request.get_json())
+    return jsonify({"task": task})
+```
+We use the id of the task that we want to update. First, it will look whether the task with provided id exists or not. If the task doesn't exist then it will return the error in JSON response indicating the task was not found, with status code `404`. If the task exists then `task.update(request.get_json())` will update the task with the new data from the request body and returns the updated task as a JSON response.
+
+### DELETE Request
+To delete a task, implement a route that handles DELETE requests:
+```python
+@app.route("/tasks/<int:task_id>", methods=["DELETE"])
+def delete_tasks(task_id):
+    task = next((item for item in tasks if item["id"] == task_id), None)
+    if task is None:
+        return jsonify({"error": "Task not found"}), 404
+    tasks.remove(task)
+    return jsonify({"result": "Task deleted"})
+```
+We use the id of the task that we want to delete. First, it will look whether the task with provided id exists or not. If the task doesn't exist then it will return the error in JSON response indicating the task was not found, with status code `404`. If the task exists then it will remove that specific task from the list of task and returns a JSON repsonse indicating task was deleted successfully.
```

Comment:
```suggestion
We use the id of the task that we want to delete. First, it will look whether the task with the provided id exists or not. If the task doesn't exist then it will return the error in JSON response indicating the task was not found, with status code `404`. If the task exists then it will remove that specific task from the list of tasks and return a JSON response indicating the task was deleted successfully.
```

Reply:
> Maybe explain the response more in depth in its own section. I guess having some understanding of the response object is useful for the user anyway and would correspond to request types

I wasn't explaining the response here, just explaining what function does and returns in the response. But I have now updated the explanation part as per the above comments.